src/libraries/System.Text.Json/src/System/Text/Json/Schema/JsonSchemaExporterOptions.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json.Nodes;
+
+namespace System.Text.Json.Schema
+{
+    /// <summary>
+    /// Configures the behavior of the <see cref="JsonSchemaExporter"/> APIs.
+    /// </summary>
+    public sealed class JsonSchemaExporterOptions
+    {
+        /// <summary>
+        /// Gets the default configuration object used by <see cref="JsonSchemaExporter"/>.
+        /// </summary>
+        public static JsonSchemaExporterOptions Default { get; } = new();
+
+        /// <summary>
+        /// Defines a callback that is invoked for every schema that is generated within the type graph.
+        /// </summary>
+        public Action<JsonSchemaExporterContext, JsonObject>? OnSchemaNodeGenerated { get; init; }
```

Comment:
@stephentoub @captainsafia how do you feel about potentially revising this API so that it becomes a transformer instead of a callback? I.e.

```C#
public Func<JsonSchemaExporterContext, JsonNode, JsonNode>? TransformSchema { get; init; }
```

Even though schema values can either be a boolean or an object, the current design is shoehorning all generated schemas through `JsonObject`  because only `JsonObject` is mutable. A transformer on the other hand lets us replace an immutable boolean schema with a different schema document.

Reply:
What would a typical delegate someone would assign look like in both cases?

Reply:
The current design is mutation based, here's [one example](https://github.com/dotnet/runtime/pull/103322/files#diff-676eabc7360cdc19fd70fb3e606f05b42a798d250f766dd69f25571e76c3d803R288-R308) picked up from the tests. A mapper-based approach would be more flexible in that you could mutate the source schema (if a source object):
```C#
static (ctx, schema) =>
{
    if (schema is JsonObject objSchema)
    {
           objSchema["$id"] = ...;
    }

    return schema;
}
```
or simply substitute `false` or `true` schemas with objects or vice versa:
```C#
static (ctx, schema) =>
{
     if (ctx.TypeInfo.Type == typeof(SomeUnsupportedType)
     {
          return (JsonNode)false;
     }

    return schema;
}
```

Reply:
The transformer-based API would make some of the modifications we make in the OpenAPI implementation a lot cleaner (e.g. ones where we replace schemas generated for well-defined types like `IFormCollection` with our own web-specific schemas).

As for as cohesiveness with the rest of the ecosystem, the OpenApiSchemaTransformer concept that we have in ASP.NET Core uses a callback based approach, although I don't think that prohibits us from following a different design here.

Calling it `Transform` might be nice as well since we use the same API name in ASP.NET Core.

Reply:
I've pushed updates converting the callback to use a transformer, which actually simplified the implementation of the exporter as well.